src/libAtomVM/nifs.c

```diff
@@ -1460,6 +1482,76 @@ static term nifs_erlang_process_info(Context *ctx, int argc, term argv[])
     return ret;
 }
 
+static int set_system_architecture(char *buf, size_t max_len)
+{
+    size_t system_name_len = strlen(SYSTEM_NAME);
+    size_t system_version_len = strlen(SYSTEM_VERSION);
+    size_t system_architecture_len = strlen(SYSTEM_ARCHITECTURE);
+    if ((system_name_len + 1 + system_version_len + 1 + system_architecture_len + 1) > max_len) {
+        return -1;
+    }
+    char *p = buf;
+    memcpy(p, SYSTEM_NAME, system_name_len);
+    p += system_name_len;
+    p[0] = '-';
+    p += 1;
+    memcpy(p, SYSTEM_VERSION, system_version_len);
+    p += system_version_len;
+    p[0] = '-';
+    p += 1;
+    memcpy(p, SYSTEM_ARCHITECTURE, system_architecture_len);
+    p += system_architecture_len;
+    p[0] = '\0';
+    return 0;
+}
+
+static size_t binary_wordsize(size_t len)
+{
+    return 2 + len / TERM_BYTES + (len % TERM_BYTES ? 1 : 0);
```

Comment:
are you looking for term_binary_data_size_in_terms ?

Reply:
Done